components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/JavaMemoryUsageLogger.java

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.healthcheck.api.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.util.EnumMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Service which logs the memory usage of carbon server.
+ */
+public class JavaMemoryUsageLogger {
+
+    private static final int BYTES_PER_MB = 1024 * 1000;
+    private static MemoryMXBean memoryMXBean;
+
+    private static final Log log = LogFactory.getLog(JavaMemoryUsageLogger.class);
+    private int delaySecondsBeforeStarting = Constants.MemoryUsageLoggerConfig.DEFAULT_WAIT_SECONDS;
+    private int loggerPrintingInterval = Constants.MemoryUsageLoggerConfig.DEFAULT_INTERVAL_SECONDS;
+    private ScheduledExecutorService memoryLogScheduler;
+    private ScheduledFuture<MemoryUsageReporter> scheduledFuture;
+
+    public JavaMemoryUsageLogger() {
+
+    }
+
+    public JavaMemoryUsageLogger(int loggerPrintingInterval) {
+
+        this.loggerPrintingInterval = loggerPrintingInterval;
+        this.delaySecondsBeforeStarting = loggerPrintingInterval * 2;
+        memoryLogScheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    public void start() {
+
+        memoryMXBean = ManagementFactory.getMemoryMXBean();
+        scheduledFuture = (ScheduledFuture<MemoryUsageReporter>) memoryLogScheduler.
+                scheduleAtFixedRate(new MemoryUsageReporter(), delaySecondsBeforeStarting, loggerPrintingInterval, TimeUnit.SECONDS);
+    }
+
+    public void stop() {
+
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(true);
+            memoryLogScheduler.shutdown();
+        }
+    }
+
+    /**
+     * Memory Usage Reporter.
+     */
+    public static class MemoryUsageReporter implements Runnable {
+
+        enum Attribute {
+            heap_memory_used_mb("Heap Memory Used"),
+            heap_memory_committed_mb("Heap Memory Committed"),
+            heap_memory_max_mb("Heap Memory Max"),
+            nonheap_memory_used_mb("Non-Heap Memory Used"),
+            nonheap_memory_committed_mb("Non-Heap Memory Committed"),
+            nonheap_memory_max_mb("Non-Heap Memory Maximum"),
+            mapped_buffers_count("Mapped Buffers Count");
+
+            public final String name;
+
+            Attribute(String name) {
+                this.name = name;
+            }
+        }
+
+        @Override
+        public void run() {
+
+            printUserReport(getMemoryUtilization());
+        }
+
+        private void printUserReport(EnumMap<Attribute, Long> attributes) {
+
+            formatAndWrite("JVM Memory Usage     Heap: used: %dM", attributes.get(Attribute.heap_memory_used_mb));
```

Comment:
BTW, can we make all into single line, so that it is easy to parse by the log analysis tools and display gadgets.

Reply:
```[2021-04-07 10:26:31,489] []  INFO {org.wso2.carbon.healthcheck.api.core.JavaMemoryUsageLogger} - JVM Memory Usage: Heap Used: 253M, Heap Committed: 696M, Heap Max: 932M, Non Heap Used: 223M, Non Heap Committed: 246M, Non Heap Max: 0M```

Reply:
Fixed

Reply:
+1